src/ex2/PokemonClient.js

```diff
@@ -0,0 +1,38 @@
+class PokemonClient {
+  async fetchPokemon(pokemonNumber) {
+    const url = "https://pokeapi.co/api/v2/pokemon/" + pokemonNumber;
+    try {
+      const response = await fetch(url);
+      if (!response.ok) {
+        return `Pokemon was not found`;
+      }
+      const json = await response.json();
+      if (json.name) {
+        return json;
+      }
+      return null; //json?.name ?? null;
+    } catch (error) {
+      alert("error, fail to fetch pokemon"); 
+      return null;
+    }
+  }
+
+  async fetchAllPokemons() {
+    const url = "https://pokeapi.co/api/v2/pokemon?limit=1126&offset=0";
```

Comment:
since the base URL for the pokemon API is always the same, it can be better practice to have it as a const.